src/connectors/gcp/auth.rs

```diff
@@ -21,28 +21,59 @@ use gouth::Token;
 use reqwest::header::HeaderMap;
 use reqwest::Client;
 
-/// returns `gouth::Token` object and bearer token header as a string
-pub(crate) fn authenticate_bearer() -> Result<(Token, String)> {
-    let token = Token::new()?;
-    let header_value_string = token.header_value()?.to_string();
-    Ok((token, header_value_string))
+pub(crate) struct GcsClient {
```

Comment:
Since this has no coverage and can only get coverage when tested against GCS in some way, i would be ok with adding `#[cfg(not(tarpaulin_include))]`, so the coverage doesnt drop overall.

Reply:
I think we also need to put this on the `impl` part. :/